api/models.py

```diff
@@ -13,104 +13,104 @@
-		ret = self.tobson()
-		ret = self._unbson(ret)
-		return json.dumps(ret)
-
-
-	@classmethod
-	def fromstore(cls, js):
-		ret = cls()
-		for key in js.keys():
-			if hasattr(ret, key):
-				it = js[key]
-				if type(it) == str:
-					it.replace("','", ' ')
-				setattr(ret, key, js[key])
-		if hasattr(ret, '_id'):
-			ret._id = ObjectId()
-		return ret
-
-	@classmethod
-	def frombson(cls, bson):
-		ret = cls()
-		for key in bson.keys():
-			if hasattr(ret, key):
-				attr = getattr(ret, key)
-				if 'frombson' in dir(attr):
-					setattr(ret, key, attr.__class__.frombson(bson[key]))
-				else:
-					setattr(ret, key, bson[key])
-
-		if '_id' in bson and bson['_id'] != None:
-			ret._id = str(bson['_id'])
-		return ret
+    def __init__(self, **kwargs):
+        self._id = None
+        self.created_on = calendar.timegm(time.gmtime())
+        self.deleted_on = None
+
+        for (arg, value) in kwargs.items():
+            self.__dict__[arg] = value
+
+    def tobson(self):
+        ret = deepcopy(self.__dict__)
+        for (key, value) in ret.items():
+            if type(value) == list and len(value) > 0:
+                one = value[0]
+                if 'tobson' in dir(one):
+                    ret_list = [x.tobson() for x in value]
+                    ret[key] = ret_list
+            elif 'tobson' in dir(value):
+                ret[key] = value.tobson()
+            else:
+                try:
+                    ret[key] = value.decode('utf-8', 'ignore') if value != None and len(value) > 0 else value
+                except:
+                    pass
+        if '_id' in ret.keys():
+            ret['_id'] = ObjectId(ret['_id'])
+        return ret
+
+    @classmethod
+    def _unbson(cls, chunk):
+        for (key, value) in chunk.items():
+            if type(value) == dict:
+                chunk[key] = unbson(value)
+            if type(value) in [list,tuple]:
+                chunk[key] = [unbson(x) for x in value]
+            if key == '_id':
+                chunk[key] = str(value)
+        return chunk
+
+    def tojson(self):
+        ret = self.tobson()
+        ret = self._unbson(ret)
+        return json.dumps(ret)
+
+
+    @classmethod
+    def fromstore(cls, js):
+        ret = cls()
+        for key in js.keys():
+            if hasattr(ret, key):
+                it = js[key]
+                if type(it) == str:
+                    it.replace("','", ' ')
+                setattr(ret, key, js[key])
+        if hasattr(ret, '_id'):
+            ret._id = ObjectId()
+        return ret
+
+    @classmethod
+    def frombson(cls, bson):
+        ret = cls()
+        for key in bson.keys():
+            if hasattr(ret, key):
+                attr = getattr(ret, key)
+                if 'frombson' in dir(attr):
+                    setattr(ret, key, attr.__class__.frombson(bson[key]))
+                else:
+                    setattr(ret, key, bson[key])
+
+        if '_id' in bson and bson['_id'] != None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)